code/game/objects/effects/mines.dm

```diff
@@ -1,162 +1,433 @@
+	if(ismob(triggerer))
+		mineEffect(triggerer)
+	visible_message(span_danger("[icon2html(src, viewers(src))] \the [src] detonates!"))
+	SEND_SIGNAL(src, COMSIG_MINE_TRIGGERED, triggerer)
 	qdel(src)
 
+//using an unarmed mine inhand deploys it.
+/obj/item/mine/attack_self(mob/user)
+	if(!armed)
+		user.visible_message("<span class='danger'>[user] deploys the [src].</span>", "<span class='notice'>You deploy the [src].</span>")
 
-/obj/effect/mine/explosive
-	name = "explosive mine"
-	var/range_devastation = 0
-	var/range_heavy = 1
-	var/range_light = 2
-	var/range_flash = 3
+		user.dropItemToGround(src)
+		anchored = TRUE
+		playsound(src, 'sound/machines/click.ogg', 60, TRUE)
 
-/obj/effect/mine/explosive/mineEffect(mob/victim)
-	explosion(loc, range_devastation, range_heavy, range_light, range_flash)
+		if(arm_delay)
+			armed = FALSE
+			update_appearance(UPDATE_ICON_STATE)
+			addtimer(CALLBACK(src, PROC_REF(now_armed)), arm_delay)
+		else
+			armed = TRUE
+			alpha = stealthpwr
+		log_admin("[key_name(user)] has placed \a [src] at ([x],[y],[z]).")
+
+//let them know the mine's done cooking
+/obj/item/mine/proc/now_armed()
+	armed = TRUE
+	alpha = stealthpwr
+	update_appearance(UPDATE_ICON_STATE)
+	playsound(src, 'sound/machines/nuke/angry_beep.ogg', 40, FALSE, -2)
+	visible_message("<span class='danger'>\The [src] beeps softly, indicating it is now active.<span>", vision_distance = COMBAT_MESSAGE_RANGE)
+
+//insert your horrible fate here
+/obj/item/mine/proc/mineEffect(mob/victim)
+	return
+
+//trying to pick up a live mine is probably up there when it comes to terrible ideas
+/obj/item/mine/attack_hand(mob/user)
+	if (armed)
+		if(disarmchance == 100)//you can just have it at this point
+			user.visible_message(span_notice("[user] carefully disarms \the [src]."), span_notice("You carefully disarm the [src]."))
+			anchored = FALSE
+			armed = FALSE
+			clicked = FALSE
+			update_appearance(UPDATE_ICON_STATE)
+			return
+		else
+			user.visible_message(span_warning("[user] extends their hand towards \the [src]!"), span_userdanger("You extend your arms to pick up \the [src], knowing that it will likely blow up when you touch it!"))
+			if(do_after(user, 25, target = src))
+				var/badchance = clamp(disarmchance / 6, 1, 100)
+				if(prob(badchance))
+					user.visible_message(span_notice("[user] picks up \the [src], which miraculously doesn't explode!"), span_notice("You pick up \the [src], which miraculously doesn't explode!"))
+					anchored = FALSE
+					armed = FALSE
+					clicked = FALSE
+					return
+				else
+					user.visible_message(span_danger("[user] attempts to pick up \the [src] only to hear a beep as it explodes in \his hands!"), span_userdanger("You attempt to pick up \the [src] only to hear a beep as it explodes in your hands!"))
+					triggermine()
+					update_appearance(UPDATE_ICON_STATE)
+					return
+	. =..()
+
+//handles disarming(and failing to disarm)
+/obj/item/mine/attackby(obj/item/I, mob/user)
+	if(armed)
```

Comment:
Save an indent level via:
`if(!armed)`
`     to_chat(user, span_notice("You hit \the [src] with [I]. Thankfully, nothing happens."))`
`return`

Reply:
true